### PR TITLE
Limits Telegun to Current Z-Level (Unless a Beacon is Emagged)

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Makes it so that the Telegun only teleports to beacons that are on its z-level. This can be extended to any beacon if you emag the target beacon. If you manage to walk off z-level and back on to set the beacon to space it will instead teleport you somewhere within 15 tiles as it would if you didnt have a beacon set.

Also reduces the Telegun's cost by two to make up for the change.

## Why It's Good For The Game
One-click permakilling Security/targets is cringe. This allows you to do silly things without perma-ending someone.

## Testing
Sent many vox on a wild space adventure.

## Changelog
:cl:
tweak: Tweaked the Telegun's targeting to only choose beacons on your current z-level unless theyre emagged.
tweak: Telegun cost reduced to 10TC
/:cl:

### CODE REVIEW

**Comment by Bm0n** [comment]
I don't think this is going to do much if you can just emag a beacon and do the same thing.

**Comment by matttheficus** [comment]
If you invest in an emag + telegun, id say its fine.

**NOTE:** Currently you can walk off zlevel to get around the check. Ill need to fix this but its too late at night for me to deal with this right now.

**Comment by OctusGit** [comment]
You know im fine with the nerf overall, but why the out of the blue 3 TC drop on the telegun? What justification does this weapon need to be even more cheaper when it still is one of the most dangerous weapons in the game.

**Comment by matttheficus** [comment]
> You know im fine with the nerf overall, but why the out of the blue 3 TC drop on the telegun? What justification does this weapon need to be even more cheaper when it still is one of the most dangerous weapons in the game.

Its a 2TC drop. It IS a massively sizeable nerf, hence the cost reduction. Using it as a "go the fuck away" button is a lot less deadly than a "go do in space" button.

If the Balance Team feels the TC cost should stay where it is, I can set it back to 12TC

**Comment by Mitchs98** [comment]
Really good change. It'll definitely cut down on the whole ease of it all. Though I'm really not sure just _how_ much it'll help, given you can just emag and chuck a beacon or ask a fellow syndie to do it, but I'm sure it'll do some good.

It's certainly better than being able to do it roundstart at any rate.

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -249,7 +249,7 @@
 
 /obj/item/projectile/energy/teleport/on_hit(atom/target, blocked = 0)
 	if(isliving(target))
-		if(teleport_target)
+		if(get_turf(teleport_target) == target.z || teleport_target.emagged)
```
Comment (Sirryan2002) [inline code/modules/projectiles/projectile/special.dm:252]: one goof here
you're trying to compare a `/turf` to a z-level value
```dm
get_turf(teleport_target) == target.z
```
you may want to first cast it to a turf var and then check the Z
```dm
var/turf/target_turf = get_turf(teleport_target)
if(target_turf.z == target.z || teleport_target.emagged)
```

Comment (S34NW) [reply code/modules/projectiles/projectile/special.dm:252]: You commented literally as I was about to, amazing :P

Comment (Sirryan2002) [reply code/modules/projectiles/projectile/special.dm:252]: we both decided to get on the review grind at the same time lmao

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -356,10 +356,10 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 
 /datum/uplink_item/jobspecific/telegun
 	name = "Telegun"
-	desc = "An extremely high-tech energy gun that utilizes bluespace technology to teleport away living targets. Select the target beacon on the telegun itself; projectiles will send targets to the beacon locked onto."
+	desc = "An extremely high-tech energy gun that utilizes bluespace technology to teleport away living targets. Select the target beacon on the telegun itself; projectiles will send targets to the beacon locked onto. Can only send targets to beacons in-sector unless they are emagged!"
 	reference = "TG"
 	item = /obj/item/gun/energy/telegun
-	cost = 12
+	cost = 10
```
Comment (S34NW) [inline code/datums/uplink_item.dm:362]: I feel like it needs to be cheaper now, to be totally honest. But I'm not a balance nerd

File: code/modules/projectiles/projectile/special.dm (hunk 1 above)
Comment (matttheficus) [reply code/modules/projectiles/projectile/special.dm]: I think i forgot to go back and fix this when Qwerty also told me of it, but it worked so it fell by the wayside. Should be fixed now.

File: code/datums/uplink_item.dm (hunk 2 above)
Comment (matttheficus) [reply code/datums/uplink_item.dm:362]: Its still rather strong as a "go the hell away" button. If it needs a further price drop, we can do so later.

File: code/modules/projectiles/projectile/special.dm [hunk 3]
```diff
@@ -240,16 +240,17 @@
...
+	var/obj/item/radio/beacon/teleport_target = null
 
 /obj/item/projectile/energy/teleport/New(loc, tele_target)
 	..(loc)
 	if(tele_target)
 		teleport_target = tele_target
 
 /obj/item/projectile/energy/teleport/on_hit(atom/target, blocked = 0)
+	var/turf/target_turf = get_turf(teleport_target)
 	if(isliving(target))
```
Comment (S34NW) [inline code/modules/projectiles/projectile/special.dm:252]:
```suggestion
	if(isliving(target) && istype(target_turf))
```
Should probably have a safety istype

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**